### ISSUE TITLE
Agents, Missions and Creating Characters

### ISSUE THREAD

**Issue by rykeon990** [evemuproject/evemu_server#140]
I just installed the server, I got everything set up smoothly but now I get a Segmentation fault upon character creation which causes the client and server to crash. The client reboots and I restart the server and am able to load the station after that. I then get a looping exception in the server terminal window when trying to contact an agent. Any thoughts?

Setup:
OS: Debian 8
MySQL 5.5
Client: Windows

**Comment by VizanArkonin** [comment]
Segfault is probably caused by server's attempt to create image in non-existent folder.
Go to the server folder (where you have **bin** and **etc** folders) and create **image_cache** folder.
Inside **image_cache** folder create following folders: **Character**, **Alliance**, **Corporation**, **InventoryType**, **Render** (last 2 are not neccessarily needed right now, so you can pass them if you want to). Thus, character portraits will go to - figures - **Character** folder.

As for agent - can you pastebin the fragment of that looping exception and link it here, please? Can't say from a hip what's wrong with it.

Cheers!

**Comment by rykeon990** [comment]
Thanks for the reply,

Those folders were created correctly, however I was running the server as root, so the API port was blocked when the server dropped root privileges. I changed the API port so it was > 1024 an it works now. I got that from here:
http://stackoverflow.com/questions/20825496/linux-denies-permission-on-socket-bind-using-boost-asio

Paste bin:
<script src="//pastebin.com/embed_js/h2YUPAWP"></script>

**Comment by comet0** [comment]
Whoops my bad, probably a race condition, on my system the API server binds before dropping root privileges. Could drop in a few seconds sleep before dropping privileges.
On 2 Feb 2016 22:06, rykeon990 <notifications@github.com> wrote:Thanks for the reply,

Those folders were created correctly, however I was running the server as root, so the API port was blocked when the server dropped root privileges. I changed the API port so it was > 1024 an it works now. I got that from here:
http://stackoverflow.com/questions/20825496/linux-denies-permission-on-socket-bind-using-boost-asio

Paste bin:
